src/js/apps/patients/worklist/worklist_app.js

```diff
@@ -90,10 +93,31 @@ const StateModel = Backbone.Model.extend({
     }));
   },
   isSelected(model) {
-    const list = this.isFlowType() ? this.get('selectedFlows') : this.get('selectedActions');
+    const list = this.getSelectedList();
 
     return list[model.id];
   },
+  getSelected(collection) {
+    if (!collection) return;
+
+    const list = this.getSelectedList();
+
+    return _.pick(list, (value, id) => {
+      return !!value && collection.get(id);
+    });
+  },
+  clearSelected() {
+    const listName = this.isFlowType() ? 'selectedFlows' : 'selectedActions';
+
+    this.set(listName, {});
+  },
+  selectAll(collection) {
+    const listName = this.isFlowType() ? 'selectedFlows' : 'selectedActions';
+
+    this.set(listName, _.reduce(collection.models, (selected, model) => {
```

Comment:
I think this could be 
```
collection.map(model=> {
return [model.id] = true;
});
```

Reply:
I think this is the same issue as above, the selected lists in the state aren't collections, they're objects:
```
selectedFlows: {
  '1': true,
  '2': false,
  '3': true
}
```
We could just do a forEach on the collection and then `list[model.id] = true`.

Reply:
I think `reduce` would do it.  you can reduce an array to a single object.